lib/diagnostics_channel.js

```diff
@@ -29,13 +30,17 @@ const { triggerUncaughtException } = internalBinding('errors');
 
 const { WeakReference } = internalBinding('util');
 
-function decRef(channel) {
-  if (channels.get(channel.name).decRef() === 0) {
-    channels.delete(channel.name);
-  }
+// Can't delete when weakref count reaches 0 as it could increment again.
+// Only GC can be used as a valid time to clean up the channels map.
+const finalizers = new SafeFinalizationRegistry((name) => {
```

Comment:
Is this actually safe? In my experience using a finalization registry ended up causing memory leaks.

Reply:
That was in older versions which this will never reach because of backport rules.